security/src/main/java/io/micronaut/security/audit/UserAuditingEntityEventListener.java

```diff
@@ -83,23 +82,27 @@ public boolean preUpdate(@NonNull EntityEventContext<Object> context) {
         };
     }
 
-    private void autoPopulateUserIdentity(@NonNull EntityEventContext<Object> context, boolean isUpdate) {
-        if (securityService.isAuthenticated()) {
-            final RuntimePersistentProperty<Object>[] applicableProperties = getApplicableProperties(context.getPersistentEntity());
-            for (RuntimePersistentProperty<Object> persistentProperty : applicableProperties) {
-                if (isUpdate) {
-                    if (!persistentProperty.getAnnotationMetadata().booleanValue(AutoPopulated.class, AutoPopulated.UPDATEABLE).orElse(true)) {
-                        continue;
-                    }
-                }
-
-                final BeanProperty<Object, Object> beanProperty = persistentProperty.getProperty();
-                getCurrentUserIdentityForProperty(beanProperty).ifPresent(identity -> context.setProperty(beanProperty, identity));
-            }
-        }
+    private void populate(@NonNull EntityEventContext<Object> context,
+                          @NonNull Class<? extends Annotation> listenerAnnotation) {
+        securityService.getAuthentication().ifPresent(authentication -> {
+            Map<Class<?>, Object> valueForType = new HashMap<>();
+            Stream.of(getApplicableProperties(context.getPersistentEntity()))
+                    .filter(persistentProperty -> shouldSetProperty(persistentProperty, listenerAnnotation))
+                    .forEach(persistentProperty -> {
+                        final BeanProperty<Object, Object> beanProperty = persistentProperty.getProperty();
+                        Object value = valueForType.computeIfAbsent(beanProperty.getType(), type -> conversionService.convert(authentication, beanProperty.getType()).orElse(null));
+                        context.setProperty(beanProperty, value);
```

Comment:
It occurs to me, this behavior is different from the original - I think we shouldn't be explicitly setting the value to null here if the conversion fails, as it could potentially overwrite an existing stored value. We should probably use `convertRequired` instead.

Reply:
I check not to be null before setting the property. I added a test as well https://github.com/micronaut-projects/micronaut-security/pull/1539/commits/a45939f898b47fe2fae46e240324fcfd963fa54a

Reply:
@sdelamo Why opt for the null check instead of switching to using `convertRequired`? 

My thinking is that failing to convert if there is a valid `Authentication` to be applied to one of the annotated fields should be an error and brought to the user's attention. It could be that they're using a custom converter and their conversion logic is wrong. Silently ignoring the failure to convert would make such a bug harder to detect.

Reply:
[I have changed to use `convertRequired`](https://github.com/micronaut-projects/micronaut-security/pull/1539/commits/66aa0ace73a5159da0f3400de328be9883d29a1f)